### PR TITLE
[REF] Use Python for denoising function

### LINKED ISSUE #37: Replace regfilt with Python code
### Summary
This is part of #2. I'm pretty sure we can use `numpy.linalg.lstsq` to handle this regression. I just have some questions about non-aggressive vs. aggressive denoising.

#### Aggressive denoising

Regress bad components out of data and retain residuals as the denoised data. That's it, right?

#### Non-aggressive denoising

Regress good components out of bad components and retain residuals, then regress updated bad components (i.e., the residuals) out of data and retain _those_ residuals as the denoised data. Is that correct?

### PR DESCRIPTION
**Pull request by tsalo**
Closes #37.

Changes proposed in this pull request:

- Replace `fsl_regfilt` with `numpy.linalg.lstsq` for denoising. I used `tedana`'s `tedort` as the basis for this.

### CODE REVIEW

File: aroma/utils.py [hunk 1]
```diff
@@ -387,60 +389,59 @@ def denoising(fsl_dir, in_file, out_dir, mel_mix, den_type, den_idx):
...
+    if motion_components_found:
+        mixing = np.loadtxt(mixing)
+        good_idx = np.setdiff1d(np.arange(mixing.shape[1]), den_idx)
+        motion_components = mixing[:, den_idx]
+        good_components = mixing[:, good_idx]
+
+        # Create a fake mask to make it easier to reshape the full data to 2D
+        img = nib.load(in_file)
+        full_mask = nib.Nifti1Image(np.ones(img.shape, int), img.affine)
+        data = masking.apply_mask(img, full_mask)  # T x S
```
Comment (eurunuela) [inline aroma/utils.py:406]: Just ran the integration test and this line raises the error:

```bash
nilearn._utils.exceptions.DimensionError: Input data has incompatible dimensionality: Expected dimension is 3D and you provided a 4D image. See http://nilearn.github.io/manipulating_images/input_output.html.

../.pyenv/versions/3.8.3/lib/python3.8/site-packages/nilearn/_utils/niimg_conversions.py:276: DimensionError
```

My guess is `full_mask` is a 4D dataset and that's causing the error.

**Review by eurunuela — CHANGES_REQUESTED** [review]
As said, the tests do not pass so I could not verify this is correct. 

Anyway, @CesarCaballeroGaudes commented in #37 that the non-aggressive denoising should be done in a different way.

File: aroma/utils.py (hunk 1 above)
Comment (tsalo) [reply aroma/utils.py:404]: Good catch. I fixed it by creating the nifti image from `img.shape[:3]`.

**Review by eurunuela — APPROVED** [review]
LGTM now! Thank you!

**Pull request merged**